heartbeat/scheduler/scheduler.go

```diff
@@ -160,11 +161,11 @@ func (s *Scheduler) WaitForRunOnce() {
 // has already stopped.
 var ErrAlreadyStopped = errors.New("attempted to add job to already stopped scheduler")
 
-type AddTask func(sched Schedule, id string, entrypoint TaskFunc, jobType string) (removeFn context.CancelFunc, err error)
+type AddTask func(sched Schedule, id string, entrypoint TaskFunc, jobType string, client *core.SyncClient) (removeFn context.CancelFunc, err error)
 
 // Add adds the given TaskFunc to the current scheduler. Will return an error if the scheduler
 // is done.
-func (s *Scheduler) Add(sched Schedule, id string, entrypoint TaskFunc, jobType string) (removeFn context.CancelFunc, err error) {
+func (s *Scheduler) Add(sched Schedule, id string, entrypoint TaskFunc, jobType string, client *core.SyncClient) (removeFn context.CancelFunc, err error) {
```

Comment:
If we do have to do this, I think it'd be cleaner to just pass in the wait method, rather than the whole client. I think the type would just be `func ()`

Reply:
Maybe it'd just be called `runOnceWait func()`, ugly, but at least we're not saying the scheduler is enmeshed with the publishing pipeline. Probably makes testing easier too.